library/src/main/java/com/google/android/exoplayer/dash/mpd/SegmentBase.java

```diff
@@ -301,7 +301,7 @@ public int getLastSegmentNum() {
         return DashSegmentIndex.INDEX_UNBOUNDED;
       } else {
         long durationMs = (duration * 1000) / timescale;
-        return startNumber + (int) (periodDurationMs / durationMs);
+        return startNumber + (int) (periodDurationMs / durationMs) - 1;
```

Comment:
Good spot. Although whilst this fixes the case where periodDurationMs divides exactly into durationMs (which is currently broken), I think it breaks the case where it does not (which is currently not broken). I think the calculation that works for both is:

```
startNumber + ceil(periodDurationMs / durationMs) - 1
```

Which in integer arithmetic is, I think:

```
startNumber + (int) ((periodDurationMs + durationMs - 1) / durationMs) - 1;
```